Gradle build doesn't work if the plugin isn't in the local repository

**Describe the bug**
When running any Gradle task, the build fails 

**Expected behavior**
Gradle build should work

**Actual behavior**
It doesn't work. The following stacktrace is shown: 

```bash
➜  code-with-quarkus ./gradlew quarkusDev

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'code-with-quarkus'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find io.quarkus:quarkus-gradle-plugin:0.23.2.
     Searched in the following locations:
       - file:/home/ggastald/.m2/repository/io/quarkus/quarkus-gradle-plugin/0.23.2/quarkus-gradle-plugin-0.23.2.pom
       - file:/home/ggastald/.m2/repository/io/quarkus/quarkus-gradle-plugin/0.23.2/quarkus-gradle-plugin-0.23.2.jar
     Required by:
         project :

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 404ms
```


**To Reproduce**
Steps to reproduce the behavior:
1. Run code.quarkus.io with this PR: https://github.com/quarkusio/code.quarkus.io/pull/163  and generate a Gradle project
2. Unzip the generated project and run `./gradlew quarkusDev`

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
```
Linux ggastald-laptop 5.2.15-200.fc30.x86_64 #1 SMP Mon Sep 16 15:17:36 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b03)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b03, mixed mode)
```
 - GraalVM version (if different from Java): Not applicable
 - Quarkus version or git rev: 0.23.2

Additional Context: 

I don't have the quarkus-gradle-plugin in my local repository

--- a/devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java
+++ b/devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java
@@ -30,18 +30,6 @@ public class GradleBuildFile extends BuildFile {
 
     public GradleBuildFile(ProjectWriter writer) throws IOException {
         super(writer, BuildTool.GRADLE);
-        if (writer.exists(SETTINGS_GRADLE_PATH)) {
-            final byte[] settings = writer.getContent(SETTINGS_GRADLE_PATH);
-            settingsContent = new String(settings, StandardCharsets.UTF_8);
-        }
-        if (writer.exists(BUILD_GRADLE_PATH)) {
-            final byte[] build = writer.getContent(BUILD_GRADLE_PATH);
-            buildContent = new String(build, StandardCharsets.UTF_8);
-        }
-        if (writer.exists(GRADLE_PROPERTIES_PATH)) {
-            final byte[] properties = writer.getContent(GRADLE_PROPERTIES_PATH);
-            propertiesContent.load(new ByteArrayInputStream(properties));
-        }
     }
 
     @Override
@@ -55,9 +43,25 @@ public class GradleBuildFile extends BuildFile {
 
     @Override
     public void completeFile(String groupId, String artifactId, String version) throws IOException {
+        init();
         completeSettingsContent(artifactId);
         completeBuildContent(groupId, version);
-        completeProperties(artifactId);
+        completeProperties();
+    }
+
+    protected void init() throws IOException {
+        if (getWriter().exists(SETTINGS_GRADLE_PATH)) {
+            final byte[] settings = getWriter().getContent(SETTINGS_GRADLE_PATH);
+            settingsContent = new String(settings, StandardCharsets.UTF_8);
+        }
+        if (getWriter().exists(BUILD_GRADLE_PATH)) {
+            final byte[] build = getWriter().getContent(BUILD_GRADLE_PATH);
+            buildContent = new String(build, StandardCharsets.UTF_8);
+        }
+        if (getWriter().exists(GRADLE_PROPERTIES_PATH)) {
+            final byte[] properties = getWriter().getContent(GRADLE_PROPERTIES_PATH);
+            propertiesContent.load(new ByteArrayInputStream(properties));
+        }
     }
 
     private void completeBuildContent(String groupId, String version) {
@@ -128,7 +132,7 @@ public class GradleBuildFile extends BuildFile {
         settingsContent = res.toString();
     }
 
-    private void completeProperties(String artifactId) {
+    private void completeProperties() {
         if (propertiesContent.getProperty("quarkusVersion") == null) {
             propertiesContent.setProperty("quarkusVersion", getPluginVersion());
         }

--- a/devtools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java
+++ b/devtools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java
@@ -40,7 +40,6 @@ public class MavenBuildFile extends BuildFile {
 
     public MavenBuildFile(ProjectWriter writer) throws IOException {
         super(writer, BuildTool.MAVEN);
-        initModel();
     }
 
     private void initModel() throws IOException {

--- a/devtools/common/src/main/resources/templates/basic-rest/kotlin/build.gradle-template.ftl
+++ b/devtools/common/src/main/resources/templates/basic-rest/kotlin/build.gradle-template.ftl
@@ -10,7 +10,7 @@ buildscript {
 }
 
 plugins {
-    id 'kotlin'
+    id 'org.jetbrains.kotlin.jvm' version "${kotlin_version}"
 }
 
 apply plugin: 'io.quarkus'
@@ -23,6 +23,7 @@ repositories {
 dependencies {
     implementation enforcedPlatform("io.quarkus:quarkus-bom:${quarkusVersion}")
     implementation 'io.quarkus:quarkus-resteasy'
+    implementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8'
 
     testImplementation 'io.quarkus:quarkus-junit5'
     testImplementation 'io.rest-assured:rest-assured'
@@ -30,3 +31,12 @@ dependencies {
 
 group '${project_groupId}'
 version '${project_version}'
+
+quarkus {
+    setOutputDirectory("$projectDir/build/classes/kotlin/main")
+}
+
+quarkusDev {
+    setSourceDir("$projectDir/src/main/kotlin")
+}
+

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/GradleBuildFileFromConnector.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/GradleBuildFileFromConnector.java
@@ -22,6 +22,8 @@ public class GradleBuildFileFromConnector extends GradleBuildFile {
 
     public GradleBuildFileFromConnector(ProjectWriter writer) throws IOException {
         super(writer);
+        // we need to initialize here since there is no other single point of entry
+        init();
     }
 
     @Override

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
@@ -1,5 +1,6 @@
 package io.quarkus.maven;
 
+import static io.quarkus.generators.ProjectGenerator.BUILD_FILE;
 import static org.fusesource.jansi.Ansi.ansi;
 import static org.twdata.maven.mojoexecutor.MojoExecutor.artifactId;
 import static org.twdata.maven.mojoexecutor.MojoExecutor.configuration;
@@ -39,6 +40,7 @@ import org.fusesource.jansi.Ansi;
 import io.quarkus.cli.commands.AddExtensionResult;
 import io.quarkus.cli.commands.AddExtensions;
 import io.quarkus.cli.commands.CreateProject;
+import io.quarkus.cli.commands.file.BuildFile;
 import io.quarkus.cli.commands.writer.FileProjectWriter;
 import io.quarkus.generators.BuildTool;
 import io.quarkus.generators.SourceType;
@@ -160,7 +162,7 @@ public class CreateProjectMojo extends AbstractMojo {
             File createdDependenciesBuildFile = new File(projectRoot, buildToolEnum.getDependenciesFile());
             File buildFile = new File(createdDependenciesBuildFile.getAbsolutePath());
             if (success) {
-                AddExtensionResult result = new AddExtensions(new FileProjectWriter(buildFile.getParentFile()), buildToolEnum)
+                AddExtensionResult result = new AddExtensions((BuildFile) context.get(BUILD_FILE))
                         .addExtensions(extensions);
                 if (!result.succeeded()) {
                     success = false;

